### PR TITLE
CBL: Astarion, the Decadent + 3 cards

### LINKED ISSUE #485: Gale's Redirection

### LINKED ISSUE #487: Astarion, the Decadent

### LINKED ISSUE #488: Legion Loyalty

### LINKED ISSUE #489: Wizards of Thay

### PR DESCRIPTION
**Pull request by Suthro**
Add scripts for 4 CBL cards:
- Astarion, the Decadent
- Gale's Redirection
- Legion Loyalty
- Wizards of Thay

closes #485 
closes #487
closes #488
closes #489

### CODE REVIEW

File: forge-gui/res/cardsfolder/upcoming/astarion_the_decadent.txt [hunk 1]
```diff
@@ -0,0 +1,15 @@
+Name:Astarion, the Decadent
+ManaCost:4 W B
+Types:Legendary Creature Vampire Elf Rogue
+PT:4/4
+K:Deathtouch
+K:Lifelink
+T:Mode$ Phase | Phase$ End of Turn | ValidPlayer$ You | TriggerZones$ Battlefield | Execute$ TrigCharm | TriggerDescription$ At the beginning of your end step, ABILITY
+SVar:TrigCharm:DB$ Charm | Choices$ DBOppLoseLife,DBGainLife | CharmNum$ 1
+SVar:DBOppLoseLife:DB$ LoseLife | ValidTgts$ Opponent | LifeAmount$ X | TgtPrompt$ Choose target opponent to lose life equal to the amount of life they lost this turn | SpellDescription$ Target opponent loses life equal to the amount of life they lost this turn.
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/astarion_the_decadent.txt:9]: Add `Feed – ` to SpellDescription

File: forge-gui/res/cardsfolder/upcoming/astarion_the_decadent.txt [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+SVar:DBGainLife:DB$ GainLife | Defined$ You | LifeAmount$ Y | SpellDescription$ You gain life equal to the amount of life you gained this turn.
+SVar:X:TargetedPlayer$LifeLostThisTurn
+SVar:Y:Count$LifeYouGainedThisTurn
+DeckHints:Type$Vampire|Elf|Rogue
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/astarion_the_decadent.txt:13]: remove this line, Forge already gets this from `Type:` line above

File: forge-gui/res/cardsfolder/upcoming/gales_redirection.txt [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+SVar:DBRoll:DB$ RollDice | Sides$ 20 | Modifier$ Y | ResultSubAbilities$ 1-14:DBMayPlay,Else:DBMayPlayWithoutCost | StackDescription$ SpellDescription | SpellDescription$ Roll a d20 and add that spell's mana value.
+SVar:DBMayPlay:DB$ Effect | StaticAbilities$ STPlay | RememberObjects$ Remembered | Duration$ Permanent | ExileOnMoved$ Exile | SubAbility$ DBCleanup
+SVar:DBMayPlayWithoutCost:DB$ Effect | StaticAbilities$ STPlayWithoutCost | RememberObjects$ Remembered | Duration$ Permanent | ExileOnMoved$ Exile | SubAbility$ DBCleanup
+SVar:STPlay:Mode$ Continuous | MayPlay$ True | MayPlayIgnoreColor$ True | EffectZone$ Command | Affected$ Card.IsRemembered | AffectedZone$ Exile | Description$ 1—14 VERT You may cast that card for as long as it remains exiled, and you may spend mana as though it were mana of any color to cast it.
+SVar:STPlayWithoutCost:Mode$ Continuous | MayPlay$ True | MayPlayWithoutManaCost$ True | EffectZone$ Command | Affected$ Card.IsRemembered | AffectedZone$ Exile | Description$ 15+ VERT You may cast that card without paying its mana cost for as long as it remains exiled.
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/gales_redirection.txt:9]: I don't think these lines should have the `number VERT` as they are just on the effects

The Effect lines should have SpellDescription$ with this so the card displays properly

File: forge-gui/res/cardsfolder/upcoming/gales_redirection.txt [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+SVar:Y:RememberedLKI$CardManaCost
+SVar:DBCleanup:DB$ Cleanup | ClearRemembered$ True
+AI:RemoveDeck:All
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/gales_redirection.txt:12]: @Agetian seems like this wouldn't be too hard for the AI to target? Is there some "counter" type AI logic already existent or does something new need to be written?

File: forge-gui/res/cardsfolder/upcoming/wizards_of_thay.txt [hunk 5]
```diff
@@ -0,0 +1,9 @@
+Name:Wizards of Thay
+ManaCost:3 U
+Types:Creature Human Wizard
+PT:3/3
+K:Myriad
+S:Mode$ ReduceCost | ValidCard$ Instant,Sorcery | Type$ Spell | Activator$ You | Amount$ 1 | Description$ Instant and sorcery spells you cast cost {1} less to cast.
+S:Mode$ CastWithFlash | ValidCard$ Sorcery | ValidSA$ Spell | Caster$ You | Description$ You may cast sorcery spells as though they had flash.
+DeckHints:Type$Instant|Sorcery|Wizard
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/wizards_of_thay.txt:8]: remove `|Wizard`

File: forge-gui/res/cardsfolder/upcoming/astarion_the_decadent.txt (hunk 1 above)
Comment (Suthro) [reply forge-gui/res/cardsfolder/upcoming/astarion_the_decadent.txt]: Added `Feed -` and `Friends -` prefixes to SpellDescriptions of both abilities

File: forge-gui/res/cardsfolder/upcoming/astarion_the_decadent.txt (hunk 2 above)
Comment (Suthro) [reply forge-gui/res/cardsfolder/upcoming/astarion_the_decadent.txt]: Removed redundant creature type deck hint line

File: forge-gui/res/cardsfolder/upcoming/gales_redirection.txt (hunk 3 above)
Comment (Suthro) [reply forge-gui/res/cardsfolder/upcoming/gales_redirection.txt]: Added spell description strings to both effects, and shifted `number VERT` prefixes to the effect spell descriptions

File: forge-gui/res/cardsfolder/upcoming/wizards_of_thay.txt (hunk 5 above)
Comment (Suthro) [reply forge-gui/res/cardsfolder/upcoming/wizards_of_thay.txt]: Removed redundant creature type deck hint

File: forge-gui/res/cardsfolder/upcoming/astarion_the_decadent.txt [hunk 6]
```diff
@@ -0,0 +1,14 @@
+Name:Astarion, the Decadent
+ManaCost:4 W B
+Types:Legendary Creature Vampire Elf Rogue
+PT:4/4
+K:Deathtouch
+K:Lifelink
+T:Mode$ Phase | Phase$ End of Turn | ValidPlayer$ You | TriggerZones$ Battlefield | Execute$ TrigCharm | TriggerDescription$ At the beginning of your end step, ABILITY
+SVar:TrigCharm:DB$ Charm | Choices$ DBOppLoseLife,DBGainLife | CharmNum$ 1
+SVar:DBOppLoseLife:DB$ LoseLife | ValidTgts$ Opponent | LifeAmount$ X | TgtPrompt$ Choose target opponent to lose life equal to the amount of life they lost this turn | SpellDescription$ Feed — Target opponent loses life equal to the amount of life they lost this turn.
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/astarion_the_decadent.txt:9]: This TgtPrompt$ probably isn't needed, personal preference if you like it though

Comment (Suthro) [reply forge-gui/res/cardsfolder/upcoming/astarion_the_decadent.txt:9]: Good to know, must have been a carry over from last PR's feedback

File: forge-gui/res/cardsfolder/upcoming/gales_redirection.txt (hunk 4 above)
Comment (Agetian) [reply forge-gui/res/cardsfolder/upcoming/gales_redirection.txt:12]: Hmm, I don't remember anything already in ChangeZoneAi in particular, it'll probably need a hook to a standard counter logic (along the lines of "if AILogic is set to CounterSpell, then run a check for the counter logic and, if it succeeds, agree to play the spell targeting whatever is on the stack"). Might look into in but need a little bit more time than currently available... :/

**Review by tool4ever — APPROVED** [review]

**Review by Northmoc — APPROVED** [review]
👍

**Pull request merged**